scripts/loader.cmd

```diff
@@ -0,0 +1,136 @@
+@REM Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+@REM SPDX-License-Identifier: Apache-2.0
+
+@echo off
+SETLOCAL EnableDelayedExpansion
+SET DIR=%~dp0
+
+@REM Get the root GG directory (generally /greengrass/v2)
+FOR %%I IN ("%DIR%\..\..\..\..") DO SET "GG_ROOT=%%~fI"
+SET LAUNCH_DIR=%GG_ROOT%\alts\current
+@REM SET CONFIG_FILE=""
+
+SET IS_SYMLINK=0
+CALL :directory_is_symlink "%GG_ROOT%\alts\new" IS_SYMLINK
+IF %IS_SYMLINK% EQU 1 (
+    CALL :directory_is_symlink "%GG_ROOT%\alts\old" IS_SYMLINK
+    IF %IS_SYMLINK% EQU 0 (
+        CALL :directory_is_symlink "%LAUNCH_DIR%" IS_SYMLINK
+        IF %IS_SYMLINK% EQU 1 (
+            CALL :flip_links "%LAUNCH_DIR%" "%GG_ROOT%\alts\old"
+        )
+    )
+    CALL :flip_links "%GG_ROOT%\alts\new" "%LAUNCH_DIR%"
+)
+
+CALL :directory_is_symlink "%GG_ROOT%\alts\broken" IS_SYMLINK
+IF IS_SYMLINK EQU 1 (
+    CALL :directory_is_symlink "%GG_ROOT%\alts\old" IS_SYMLINK
+    IF IS_SYMLINK EQU 1 (
+        CALL :flip_links "%GG_ROOT%\alts\old" "%LAUNCH_DIR%"
+    )
+)
+
+CALL :directory_is_symlink "%GG_ROOT%\alts\old" IS_SYMLINK
+IF IS_SYMLINK EQU 1 (
+    CALL :directory_is_symlink "%LAUNCH_DIR%" IS_SYMLINK
+    IF IS_SYMLINK EQU 0 (
+        CALL :flip_links "%GG_ROOT%\alts\old" "%LAUNCH_DIR%"
+    )
+)
+
+@REM EXIST works for files, directories, and symlink dirs
+IF NOT EXIST %LAUNCH_DIR% (
+    ECHO FATAL: No Nucelus found!
+    EXIT /B 1
+)
+
+@REM Get JVM_OPTIONS from launch.params if it exists
+IF EXIST %LAUNCH_DIR%\launch.params (
+    FOR /F "delims=" %%A IN (%LAUNCH_DIR%\launch.params) DO SET JVM_OPTIONS=%%A
+)
+
+SET JVM_OPTIONS=%JVM_OPTIONS% -Droot="%GG_ROOT%"
+SET OPTIONS=--setup-system-service false
+
+ECHO JVM options: %JVM_OPTIONS%
+ECHO Nucleus options: %OPTIONS%
+SET /A MAX_RETRIES=3
+@REM Attempt to start the nucleus 3 times
+FOR /L %%i IN (1,1,%MAX_RETRIES%) DO (
+    java -Dlog.store=FILE %JVM_OPTIONS% -jar "%LAUNCH_DIR%\distro\lib\Greengrass.jar" %OPTIONS%
+    SET KERNEL_EXIT_CODE=%ERRORLEVEL%
+
+    IF KERNEL_EXIT_CODE EQU 0 (
+        ECHO Restarting Nucleus
+        %LAUNCH_DIR%\distroy\bin\loader.cmd
+    ) ELSE (
+    IF KERNEL_EXIT_CODE EQU 100 (
```

Comment:
What do these error codes translate to?

Reply:
the kernel has a `shutdown()` method that accepts these as arguments
https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/36f8d2faf5439e52a4e3c5f739c1df8971e5a0ab/src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapSuccessCode.java#L9-L11